### PR TITLE
Accurately update state if prepare() partially fails

### PR DESCRIPTION
**Pull request by mheon**
We are seeing some issues where, when part of prepare() fails
(originally noticed due to a bad static IP), the other half does
not successfully clean up, and the state can be left in a bad
place (not knowing about an active SHM mount for example).

### CODE REVIEW

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -66,17 +66,55 @@ func (c *Container) prepare() (err error) {
 		defer wg.Done()
 		// Set up network namespace if not already set up
 		if c.config.CreateNetNS && c.state.NetNS == nil && !c.config.PostConfigureNetNS {
-			saveNetworkStatus = true
 			netNS, networkStatus, createNetNSErr = c.runtime.createNetNS(c)
+
+			tmpStateLock.Lock()
```
Comment (baude) [inline libpod/container_internal_linux.go]: Is there anyway we can avoid a lock in the main path?

Comment (mheon) [reply libpod/container_internal_linux.go]: We only take it after the work is done - so this shouldn't impact execution time in a significant way, we just hold it during the write to the struct

Comment (baude) [reply libpod/container_internal_linux.go]: sorry, was confusing with a container lock.

**Comment by baude** [comment]
LGTM

File: libpod/networking_linux.go [hunk 2]
```diff
@@ -90,13 +90,16 @@ func (r *Runtime) configureNetNS(ctr *Container, ctrNS ns.NetNS) ([]*cnitypes.Re
 }
 
 // Create and configure a new network namespace for a container
-func (r *Runtime) createNetNS(ctr *Container) (ns.NetNS, []*cnitypes.Result, error) {
+func (r *Runtime) createNetNS(ctr *Container) (n ns.NetNS, q []*cnitypes.Result, err error) {
```
Comment (giuseppe) [inline libpod/networking_linux.go]: could we drop the named return values?  They don't seem used in the function

**Comment by umohnani8** [comment]
LGTM

**Comment by mheon** [comment]
Tests are good, anyone want to /lgtm? @rhatdan @baude @TomSweeneyRedHat @umohnani8

**Pull request merged**